### ISSUE TITLE
How should NSFileManager be initialized?

### ISSUE THREAD

**Issue by timkettering** [ZipArchive/ZipArchive#198]
Opening this issue to bring up the question of how ZipArchive should correctly initialize `NSFileManager`.

From the Apple Developer docs

>>> In iOS and Mac OS X v 10.5 and later you should consider using [[NSFileManager alloc] init] rather than the singleton method defaultManager. Instances ofNSFileManager are considered thread-safe when created using [[NSFileManager alloc] init].

SSZipArchive uses the `defaultManager` method. 

Hopefully someone who knows this stuff better than I do can affirmatively answer.

cheers!

**Comment by jhudsonWA** [comment]
Good question. I know there was another issue about the project possibly not working well in a async environment. Maybe this is the cause. I don't know enough in this area to give an opinion either way. I've always used the defaultManager in all my code, but mostly because I didn't know better.

**Comment by timkettering** [comment]
Yes, one of my projects relied on downloads and unzipping on an async background thread.  I was encountering random crashes when multiple operations were running at the same time and while I cannot conclusively say it's due to this, but I went and changed all NSFileManager instances to the thread safe one and it seems to have resolved the issue.   The only caveat is that I also made other unrelated changes to background processing.  So I can't confirm.  Although later on when I have time, I can try reverting the changes and making this change in insolation for testing/confirmation.
